engine/src/main/java/org/pentaho/di/pan/delegates/PanTransformationDelegate.java

```diff
@@ -0,0 +1,207 @@
+  );
+
+  public PanTransformationDelegate( LogChannelInterface log ) {
+    this( log, null );
+  }
+
+  public PanTransformationDelegate( LogChannelInterface log, Repository repository ) {
+    this.log = log;
+    this.repository = repository;
+  }
+
+  /**
+   * Execute a transformation with the specified configuration.
+   *
+   * @param trans              the transformation
+   * @param executionConfiguration the execution configuration
+   * @param arguments              command line arguments
+   * @return the execution result
+   * @throws KettleException if execution fails
+   */
+  public Result executeTransformation( final Trans trans,
+                                       final TransExecutionConfiguration executionConfiguration,
+                                       final String[] arguments ) throws KettleException {
+
+    TransMeta transMeta = trans.getTransMeta();
+    if ( transMeta == null ) {
+      throw new KettleException( BaseMessages.getString( pkg, "PanTransformationDelegate.Error.TransMetaNull" ) );
+    }
+
+    // Set repository and metastore information in both the exec config and the metadata
+    transMeta.setRepository( repository );
+    transMeta.setMetaStore( MetaStoreConst.getDefaultMetastore() );
+    executionConfiguration.setRepository( repository );
+
+    // Set the run options
+    transMeta.setClearingLog( executionConfiguration.isClearingLog() );
+    transMeta.setSafeModeEnabled( executionConfiguration.isSafeModeEnabled() );
+    transMeta.setGatheringMetrics( executionConfiguration.isGatheringMetrics() );
+
+    // Call extension points
+    ExtensionPointHandler.callExtensionPoint( log, KettleExtensionPoint.SpoonTransMetaExecutionStart.id, transMeta );
+    ExtensionPointHandler.callExtensionPoint( log, KettleExtensionPoint.SpoonTransExecutionConfiguration.id, executionConfiguration );
+
+    try {
+      ExtensionPointHandler.callExtensionPoint( log, KettleExtensionPoint.SpoonTransBeforeStart.id, new Object[] {
+        executionConfiguration, transMeta, transMeta, repository
+      } );
+    } catch ( KettleException e ) {
+      log.logError( e.getMessage(), transMeta.getFilename() );
+      throw e;
+    }
+
+    // Apply parameters
+    Map<String, String> paramMap = executionConfiguration.getParams();
+    for ( Map.Entry<String, String> entry : paramMap.entrySet() ) {
+      transMeta.setParameterValue( entry.getKey(), Const.NVL( entry.getValue(), "" ) );
+    }
+    transMeta.activateParameters();
+
+    // Set the log level
+    if ( executionConfiguration.getLogLevel() != null ) {
+      transMeta.setLogLevel( executionConfiguration.getLogLevel() );
+    }
+
+    // Determine execution type and execute accordingly
+    return executeBasedOnConfiguration( transMeta, executionConfiguration, arguments );
+  }
+
+  /**
+   * Execute transformation based on the execution configuration type.
+   */
+  private Result executeBasedOnConfiguration( TransMeta transMeta,
```

Comment:
Looking at the test code, I don't see where you verify that the correct mapping is happening for example
executionConfiguration.isExecutingRemotely() ->  RemoteTransExecutorService#execute

There isn't a mockito.verify() for execute() for an instance of a  RemoteTransExecutorService class

I see that

Reply:
fixed